ROC.py

```diff
@@ -0,0 +1,124 @@
+"""
+@auther: Tziporah
+"""
+import warnings
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from sklearn.metrics import accuracy_score, roc_curve, auc, RocCurveDisplay, confusion_matrix, f1_score
+
+
+warnings.simplefilter(action='ignore', category=FutureWarning)
+warnings.simplefilter(action='ignore', category=pd.errors.PerformanceWarning)
+plt.style.use('seaborn-v0_8')
+
+
+class ROCMetrics:
+    def __init__(self, y_true: pd.Series, p_pred: pd.Series):
+        self.y_true = y_true
+        self.p_pred = p_pred
+
+    def conf_matrix(self, y_pred: pd.Series) -> np.array:
+        return confusion_matrix(self.y_true, y_pred)
+
+    @staticmethod
+    def sensitivity(conf_matrix: np.array) -> float:
+        tp = conf_matrix[1][1]
+        fn = conf_matrix[1][0]
+
+        with np.errstate(divide='ignore', invalid='ignore'):
+            sens = np.true_divide(tp, tp + fn)
+            if sens == np.inf:
+                sens = 0
+
+        return sens
+
+    @staticmethod
+    def specificity(conf_matrix: np.array) -> float:
+        tn = conf_matrix[0][0]
+        fp = conf_matrix[0][1]
+
+        with np.errstate(divide='ignore', invalid='ignore'):
+            spec = np.true_divide(tn, tn + fp)
+            if spec == np.inf:
+                spec = 0
+
+        return spec
+
+    @staticmethod
+    def precision(conf_matrix: np.array) -> float:
+        tp = conf_matrix[1][1]
+        fp = conf_matrix[0][1]
+
+        with np.errstate(divide='ignore', invalid='ignore'):
+            prec = np.true_divide(tp, tp + fp)
+            if prec == np.inf:
+                prec = 0
+
+        return prec
+
+    def accuracy(self, y_pred: pd.Series) -> float:
+        return accuracy_score(self.y_true, y_pred)
+
+    def f1_score(self, y_pred: pd.Series) -> float:
+        return f1_score(self.y_true, y_pred)
+
+    def threshold_matrix(self, step_size: float) -> pd.DataFrame:
```

Comment:
Nice. For ways to speed up, can def mention making this loop multi-threaded, well multiprocessed to get around python GIL but you know what I  mean.